### ISSUE TITLE
Opening changes / diff view results in lots of errors for ","ref":"~"} file

### ISSUE THREAD

**Issue by jhasse** [MaskRay/vscode-ccls#50]
When I open the diff view of any file, I get lots of errors for a non existing file (with `","ref":"~"}` appended). Opening the same file normally works fine.

It looks like this:

![bildschirmfoto von 2019-02-27 11-29-25](https://user-images.githubusercontent.com/80071/53484048-2bb63900-3a83-11e9-92e5-b64e80b2643b.png)

Seems like the path doesn't get passed to ccls correctly?

**Comment by MaskRay** [comment]
The language server protocol uses `LocationUri` to denote a document, which is usually `file://xxxx`. What protocol does the diff view use? If it does not use `file://`, I think ccls shouldn't be enabled in this case.

If it does use `file://`, it depends if the buffer content matches the file contents on the disk. In the following cases, ccls shouldn't be enabled.

```
// virtual line due to the diff view
// first line whose line number is 2 due to the existence of the virtual line
```

**Comment by jhasse** [comment]
No idea. But I remember having the same problem with cquery a long time ago, but it was fixed at some point.

**Comment by BlueHeaven1969** [comment]
Yes, this bugs me too.  If the diff is from Git, then the URI starts with git: which is the source of the problem, I think.

Cquery fixed it in:
https://github.com/cquery-project/cquery/commit/5be919962e7dda0ed4a0fc0a68061d00d2916433

**Comment by Yanpas** [comment]
I use SVN...

**Comment by BlueHeaven1969** [comment]
Given the above cquery checkin, I just put:
`  if (path.rfind("git:", 0) == 0) return;`
at line 44 in src/messages/textDocument_did.cc and that seemed to work just fine.

For SVN, see what the URI is in the problems panel.  It is probably something like svn: and you can use that.

**Comment by MaskRay** [comment]
`return StartsWith(path, "git:");`

Can that be filtered on the client side?

**Comment by BlueHeaven1969** [comment]
I agree this is not something that should be handled on the server.

Is there a way to filter these using a blacklist in the extension?  I tried filtering using ccls.index.blacklist but it didn't seem to work.
